File: queue/queue.js
class Queue {
    constructor(first_element=null) {
        this.head = first_element
        this.tail = null
    }

    push(data) {
        if (this.head == null) {
            this.head = data
            return null
        }
        if (this.tail) {
            this.tail.push(data)
        } else {
            this.tail = new Queue(data)
        }
    }

    pop() {
        var temp = this.head
        if (this.tail) {
            this.head = this.tail.head
            this.tail = this.tail.tail
        } else {
            this.head = null
        }
        return temp
    }
}

var a = new Queue()
a.push(1)
a.push(2)
console.log(a.pop())
console.log(a.pop())
console.log(a.pop())

File: queue/queue.py
class Queue:
    def __init__(self, first_element=None):
        self.head = first_element
        self.tail = None
    
    def push(self, data):
        if self.head is None:
            self.head = data
            return None
        if self.tail:
            self.tail.push(data)
        else:
            self.tail = Queue(data)

    def pop(self):
        temp = self.head
        if self.tail:
            self.head = self.tail.head
            self.tail = self.tail.tail
        else:
            self.head = None
        return temp

a = Queue()
a.push(1)
a.push(2)
print(a.pop())
print(a.pop())
print(a.pop())
